### ISSUE TITLE
Add support for React Native

### ISSUE THREAD

**Issue by rturk** [msintuneappsdk/ms-intune-app-sdk-ios#21]
Please Add support for React Native based Apps

**Comment by stuartbrameld** [comment]
@Kyle-Reis this would be super helpful for us as well, for an internal app, though we may put it in the public app store as well

**Comment by Kyle-Reis** [comment]
Hey @rturk and @stuartbrameld, I would recommend taking a look at our [app wrapping tool](https://github.com/msintuneappsdk/intune-app-wrapping-tool-ios). It is intended for internal LOB apps, but some customers have had sucess submitting to the App Store.

**Comment by rturk** [comment]
@Kyle-Reis we'll distribute in the app store. this is our project https://feedback.house/. most of our customers already log via SAML, however customers with more strict security requirements asked us to consider Intune as additional security layer

**Comment by stuartbrameld** [comment]
@rturk we just across across this repo that may be of interest https://github.com/VarianMobileARIA/react-native-ms-intune-mam

**Comment by stuartbrameld** [comment]
@Kyle-Reis do you know of others that have successfully used app protection policies (e.g. data wipe https://docs.microsoft.com/en-us/intune/app-sdk#remotely-wipe-corporate-data) with React Native by creating a bridge to the IOS/Android SDKs? App wrapping feels a little clunky in comparison with the SDK? Many thanks for your help.

**Comment by Kyle-Reis** [comment]
@stuartbrameld, I'm not aware of any such apps. If your app only supports a single corporate identity, app wrapping might make sense. You can see a comparative list of supported features/scenarios [here](https://docs.microsoft.com/en-us/intune/apps-prepare-mobile-application-management#feature-comparison). 

Notes: 

"Replace built-in app PIN with Intune PIN" - Just to be sure there is no confusion here, Intune PIN policy is fully supported in wrapped apps. The app wrapper is typically used by customers who don't have source code access or technical know-how to modify their LOB apps themselves (as they may have been developed by an external vendor), so this line is to set the expectation with those customers that users may be prompted for PIN twice if Intune PIN policy is set and the app has its own built-in PIN prompt.

Selective wipe - For wrapped apps, when a selective wipe is received, the Intune SDK will wipe the entire application directory, as we are operating under the assumption that it is a single identity app and all data is corporate data.

**Comment by stuartbrameld** [comment]
@Kyle-Reis thanks Kyle, I'll let you know what we decide and how we get on

**Comment by Kyle-Reis** [comment]
Hey folks, just checking in to verify that you're unblocked before closing out this issue.

**Comment by msintuneappsdk** [comment]
Hi all, the Intune SDK development team actively tests and maintains support for apps built with the native Android, iOS (Obj-C, Swift), Xamarin, Xamarin.Forms, and Cordova platforms. While some customers have had success with Intune SDK integration with other platforms such as React Native and NativeScript, we do not provide explicit guidance or plugins for app developers using anything other than our supported platforms.

Closing this issue.

**Comment by eyast** [comment]
Hi - is there a recent supportability statement to the above? thanks

**Comment by sdevansh96** [comment]
Hello,

Is their any support for React Native based Apps?

**Comment by sdevansh96** [comment]
> Hi all, the Intune SDK development team actively tests and maintains support for apps built with the native Android, iOS (Obj-C, Swift), Xamarin, Xamarin.Forms, and Cordova platforms. While some customers have had success with Intune SDK integration with other platforms such as React Native and NativeScript, we do not provide explicit guidance or plugins for app developers using anything other than our supported platforms.
> 
> Closing this issue.

Can you please add support for React Native?

**Comment by ansarikhurshid786** [comment]
I am also looking for support for react native. any help

**Comment by stellarpro** [comment]
Same, multiple high profile clients have now requested MAM support, can we have SDK for react native apps please?

**Comment by sirtobey** [comment]
Reopening this, as it's stale, and we need the support for ReactNative. Cheers.

**Comment by eikhunter** [comment]
+1 to opening this issue, would love to see a RN SDK for this

**Comment by StephenStrickland** [comment]
Would love to see this

**Comment by ACivilise** [comment]
We've successfully integrated the Intune SDK with our React Native project, allowing for effective operation under conditional access policies. Here's how we achieved this:

- For the Android component of our project, we incorporated the Android SDK in the same way as outlined in this existing project: https://github.com/msintuneappsdk/Taskr-Sample-Intune-Android-App.

- To enable cross-application SSO, we adjusted our MSAL configuration. We set the 'brokerredirecturi_registered' boolean to 'true'. Reference can be found here: https://github.com/stashenergy/react-native-msal/blob/14ecbf8159dd0855c4b2137e933597a1d2c4649f/src/types.ts#L90.

> [!TIP]
> For more information on how cross-app SSO works, consider visiting: https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-android-single-sign-on."
